src/rgw/rgw_rest_swift.cc

```diff
@@ -1491,7 +1491,10 @@ void RGWInfo_ObjStore_SWIFT::list_swift_data(Formatter& formatter,
 
   string ceph_version(CEPH_GIT_NICE_VER);
   formatter.dump_string("version", ceph_version);
-  formatter.dump_int("max_meta_name_length", 81);
+
+  const size_t meta_name_limit = g_conf->osd_max_attr_name_len
+    - strlen(RGW_ATTR_PREFIX);
+  formatter.dump_int("max_meta_name_length", meta_name_limit);
 
```

Comment:
@pritha-srivastava: on my environment the max length allowed is exactly 80 bytes (tested on RadosGW without the commits applied):
```
rgw$ curl -i "$publicURL/cont/" -X POST -H "X-Auth-Token: $token" -H "X-Container-Meta-`python -c 'print("a" * 81)'`: OK"
HTTP/1.1 500 Internal Server Error
Content-Length: 12
X-Trans-Id: tx00000000000000000000f-00583446c8-1016-default
Accept-Ranges: bytes
Content-Type: text/plain; charset=utf-8
Date: Tue, 22 Nov 2016 13:23:20 GMT

UnknownError

rgw$ curl -i "$publicURL/cont/" -X POST -H "X-Auth-Token: $token" -H "X-Container-Meta-`python -c 'print("a" * 80)'`: OK"
HTTP/1.1 204 No Content
X-Trans-Id: tx000000000000000000010-0058344ad0-1016-default
Content-Type: text/plain; charset=utf-8
Date: Tue, 22 Nov 2016 13:40:32 GMT

```

Output from `ngrep`:
```
T 127.0.0.1:54969 -> 127.0.0.1:8000 [AP]
POST /v1/KEY_2987047fc47840058e89e6182e0d96c3/cont/ HTTP/1.1.
User-Agent: curl/7.35.0.
Host: 127.0.0.1:8000.
Accept: */*.
X-Auth-Token: ee4025e5ca334747a0eaaa22e9b45e94.
X-Container-Meta-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa: OK.
.

##
T 127.0.0.1:8000 -> 127.0.0.1:54969 [AP]
HTTP/1.1 500 Internal Server Error.
Content-Length: 12.
X-Trans-Id: tx00000000000000000000f-00583446c8-1016-default.
Accept-Ranges: bytes.
Content-Type: text/plain; charset=utf-8.
Date: Tue, 22 Nov 2016 13:23:20 GMT.
.

##
T 127.0.0.1:8000 -> 127.0.0.1:54969 [AP]
UnknownError

########
T 127.0.0.1:54971 -> 127.0.0.1:8000 [AP]
POST /v1/KEY_2987047fc47840058e89e6182e0d96c3/cont/ HTTP/1.1.
User-Agent: curl/7.35.0.
Host: 127.0.0.1:8000.
Accept: */*.
X-Auth-Token: ee4025e5ca334747a0eaaa22e9b45e94.
X-Container-Meta-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa: OK.
.

##
T 127.0.0.1:8000 -> 127.0.0.1:54971 [AP]
HTTP/1.1 204 No Content.
X-Trans-Id: tx000000000000000000010-0058344ad0-1016-default.
Content-Type: text/plain; charset=utf-8.
Date: Tue, 22 Nov 2016 13:40:32 GMT.
.

```
Dump from the messenger:
```
2016-11-22 14:50:00.568233 7fc408817700  1 -- 127.0.0.1:0/3331651425 --> 127.0.0.1:6800/23074 -- osd_op(unknown.0.0:731 3.b843b263 .bucket.meta.298704
7fc47840058e89e6182e0d96c3:cont:89de0a15-d949-4289-99af-b81c2ac46155.4118.1 [delete,create 0~0,call version.check_conds,call version.set,writefull 0~4
35,setxattr user.rgw.acl (379),setxattr user.rgw.x-amz-meta-aaaaaaaaaaaaaaaaaaaaaa (3),setxattr user.rgw.x-amz-meta-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa (3),setxattr user.rgw.x-amz-meta-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
aaaaaaaaaaa (3)] snapc 0=[] ondisk+write+known_if_redirected e24) v7 -- 0x7fc4462fbe40 con 0
```